File: arrays/array.go
package main

import (
    "fmt"
)

const (
    arrSize = 50
    arrCapacity = 50
)

func main () {
    arr := make([]int, arrSize, arrCapacity)

    fmt.Printf("%v\n", arr)
    fmt.Printf("%d\t%d\n", len(arr), cap(arr))

    arr = append(arr, 100)

    fmt.Printf("%v\n", arr)
    fmt.Printf("%d\t%d\n", len(arr), cap(arr))

    return;
}

File: arrays/array.cpp

#include <iostream>

#define arr_size 10

int main (void) {
    int *arr = new int[arr_size];

    for (int i = 0; i < arr_size; i++) {
        std::cout << arr[i] << "\t";
    }

    std::cout << std::endl;

    for (int i = 0; i < arr_size; i++) {
        std::cout << arr + i << "\t";
    }

    return 0;
}
